deepchem/transfer_learning/chemnet.py

```diff
@@ -0,0 +1,217 @@
+"""
+General purpose transfer learning script. Currently supports only mode transfer
+(classification to regression and vice-versa) and different multi-tasking options.
+For more advanced finetuning, use the load_from_pretrained function as part of
+deepchem.models.KerasModel
+"""
+
+__author__ = "Vignesh Ram Somnath"
+__license_ = "MIT"
+
+import numpy as np
+import tensorflow as tf
+import os
+import logging
+import deepchem
+import tempfile
+
+from deepchem.models.tensorgraph.optimizers import RMSProp
+from deepchem.molnet.load_function.chembl25_datasets import chembl25_tasks
+from deepchem.models.callbacks import ValidationCallback, EarlyStoppingCallBack
+from deepchem.transfer_learning import get_model_str
+
+logger = logging.getLogger(__name__)
+
+
+def pretrain_model(chembl_config,
+                   model_class,
+                   model_hparams,
+                   train_config,
+                   metric_fn=None,
+                   **kwargs):
+  """Pretraining the model on ChEMBL dataset.
+
+    Parameters
+    ----------
+    chembl_config: dict,
+        Arguments used for loading ChEMBL dataset
+    model_class: dc.models.KerasModel
+        Model class
+    model_hparams: dict,
+        Dictionary of model arguments
+    train_config: dict,
+        Dictionary of training options
+    metric_fn: deepchem.metrics.Metric, default None
+        Metric used for evaluating training progress
+    """
+  if metric_fn is None:
+    metric_fn = deepchem.metrics.rms_score
+  metric = deepchem.metrics.Metric(
+      metric_fn, task_averager=np.mean, mode="regression", verbose=False)
+
+  tasks, dataset, transformers = deepchem.molnet.load_chembl25(**chembl_config)
+  train, valid, test = dataset
+
+  model_hparams = update_hparams(model_hparams)
+  if model_hparams.get('model_dir', None) is None:
+    model_dir = tempfile.mkdtemp()
+  else:
+    model_dir = model_hparams.get('model_dir')
+
+  n_samples = len(train.w)
+  batch_size = model_hparams.get('batch_size')
+  nb_epoch = train_config.get('nb_epoch', 50)
+  patience_early_stopping = train_config.get('patience_early_stopping', 10)
+
+  if n_samples % batch_size == 0:
+    steps_per_epoch = n_samples // batch_size
+  else:
+    steps_per_epoch = 1 + (n_samples // batch_size)
+
+  callbacks = [
+      ValidationCallback(
+          dataset=valid, metrics=[metric], interval=steps_per_epoch),
+      EarlyStoppingCallBack(
+          dataset=valid,
+          metric='loss',
+          interval=steps_per_epoch,
+          patience=patience_early_stopping,
+          save_dir=model_dir)
+  ]
+
+  model = model_class(n_tasks=len(tasks), **model_hparams)
+  model.fit(train, nb_epoch, checkpoint_interval=0, callbacks=callbacks)
+  test_scores = model.evaluate(test, metrics=[metric], [])
+  logger.info("Test-{}: {}".format(metric.name, test_scores[metric.name]))
+  return model
+
+
+def finetune_model(dataset_name,
```

Comment:
Could we generalize finetune to work on a user specified dataset? This would be a really useful features for users working with live small datasets

Reply:
Yes the arguments should be train, valid, test datasets then instead of `dataset_name` and `config`.